ROADMAP.md

```diff
@@ -0,0 +1,135 @@
+# ROADMAP
+
+See [README.md](README.md) for developer getting start instructions.
+
+## Minimal complete pipeline
+
+* `heareval/baseline.py` - Simple unlearned embedding algorithm
+following the API.
+
+* `heareval/tasks/` - One time preprocessing of tasks into train +
+test WAVs and `train/test.csv`.
+
+* `heareval/predict/` - [not implemented] For each task, map from
+API-compatible embeddings to `predicted-test.csv`. (This involves
+a linear model for most task types.)
+
+* `heareval/evaluate/` - Given task type, `test.csv`, and
+`predicted-test.csv` output evaluation scores.
+(Note that we can implement this now just by creating random `test.csv`
+and `predicted-test.csv` files, Christian is starting this task.)
+
+## Task Types + Prediction
+
+See [heareval/tasks/README.md](heareval/tasks/README.md) for more
+details.
+
+Ideally, `predicted-test.csv` would have identical format as
+`test.csv`. In practice, for AUC which depends upon the predicted
+score, we might need to change the format slightly.
```

Comment:
I could see one central csv with one column that just indicates the kind of task; and the task-inappropriate data could be filled with NaN's or something. As you wish.

Reply:
@maxsolomonhenry it gets a little more tricky than this in the multi-label scenario. I was chatting with @csteinmetz1 about this a bit.

Because for the predicted-test.csv, we could conceivably have a unbounded float score (or [0, 1] probability) for all---say---128 midi notes.